src/include/sof/spinlock.h

```diff
@@ -93,21 +93,21 @@ extern uint32_t lock_dbg_user[DBG_LOCK_USERS];
 		} \
 		if (__tries == 0) { \
 			trace_lock_error("DED"); \
-			trace_lock_error("line: %d", __LINE__); \
+			trace_lock_error("line: %d", line); \
```

Comment:
Do we need to pass in line here ? This is a macro so __LINE__ should resolve to the callers line.

Reply:
I don't think the line within the spinlock_* functions is of interest, but the line of the caller of spinlock functions. This is a macro called from a function, not directly from client code. That's why I passed it as parameter.